runelite-client/src/main/java/net/runelite/client/ui/overlay/infobox/Timer.java

```diff
@@ -39,14 +39,19 @@
 	private final Duration duration;
 
 	public Timer(long period, ChronoUnit unit, BufferedImage image, Plugin plugin)
+	{
+		this(Instant.now(), period, unit, image, plugin);
+	}
+
+	public Timer(Instant startTime, long period, ChronoUnit unit, BufferedImage image, Plugin plugin)
```

Comment:
This constructor seems unnecessary? You should be able to calculate out the length yourself when creating the timer.

Reply:
It's there so the timer consistently becomes red at the same number every time (which I think is at 10%). If the timer starts at 5 mins (which it can do upon login if the user was already there for 5 min earlier) the text would be red at 30 sec instead of 60 sec.

Reply:
Oh. It might be better to make the warning time modifiable then (overriding a method?)

Reply:
I agree with @Adam- , everywhere else we simply override the methods when needed. Simply override getColor